### PR TITLE
[FLINK-14469][python] Drop Python 2 support for PyFlink

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request drops Python 2 support.*

## Brief change log

  - *Remove the CI for Python 2.7*
  - *Throw an exception if Python below 3.5 is used*
  - *Clean up code specific for Python 2.x*

## Verifying this change

This change is a code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-python/setup.py [hunk 1]
```diff
@@ -24,8 +24,8 @@
 
 from setuptools import setup
 
-if sys.version_info < (2, 7):
-    print("Python versions prior to 2.7 are not supported for PyFlink.",
+if sys.version_info < (3, 5):
```
[the remark this answers is not in the document]
Comment (dianfu) [reply flink-python/setup.py]: Good catch! I think all Python version greater than 3.5 should be supported. So we should support Python 3.8 here. What do you think?

**Comment by dianfu** [comment]
@hequn8128 Thanks a lot for your review. Updated the PR.

File: docs/flinkDev/building.md [hunk 2]
```diff
@@ -65,6 +65,8 @@ Then go to the root directory of flink source code and run this command to build
 cd flink-python; python3 setup.py sdist bdist_wheel
 {% endhighlight %}
 
+<span class="label label-info">Note</span> Python 3.5+ is required to build PyFlink.
```
Comment (hequn8128) [inline docs/flinkDev/building.md]: Building PyFlink requires Python 3.5 or higher ?

**Comment by hequn8128** [comment]
@dianfu The PR LGTM, will address the comments above during merging...

**Pull request closed without merging**